src/reporters/lang/en.js

```diff
@@ -174,6 +174,11 @@ const messages = {
   workspaceNameMandatory: 'Missing name in workspace at $0, ignoring.',
   workspaceNameDuplicate: 'There are more than one workspace with name $0',
 
+  cacheFolderSkipped: 'Skipping preferred cache folder $0 because it is not writable.',
+  cacheFolderMissing:
+    "Yarn hasn't been able to find a cache folder. Please use an explicit --cache-folder option to tell it what location to use, or make one of the preferred locations writable.",
```

Comment:
>Yarn hasn't been able to find a cache folder.

find a cache folder it can use

>Please use an explicit --cache-folder option

use *the* explicit --cache-folder option